analysis.py

```diff
@@ -1,160 +1,536 @@
-import os.path
+import os
 import traceback
+import pandas as pd
+import numpy as np
+from pathlib import Path
+from copy import deepcopy
+import json
+from datetime import datetime
+import typing
+import pathos as pa
+from functools import partial
+import pycountry
 
-from pipeline.direct.calc_yearset import nccs_yearsets_simple
-# from utils.s3client import download_from_s3_bucket, upload_to_s3_bucket
-from pipeline.direct.direct import get_sector_exposure, nccs_direct_impacts_list_simple
+from climada.engine import Impact
+
+from pipeline.direct.direct import get_sector_exposure, nccs_direct_impacts_simple
+from pipeline.direct.calc_yearset import yearset_from_imp, combine_yearsets
 from pipeline.indirect.indirect import dump_direct_to_csv, dump_supchain_to_csv, supply_chain_climada
 from utils import folder_naming
+from utils.s3client import upload_to_s3_bucket, file_exists_on_s3_bucket, download_from_s3_bucket
 
 
-def calc_supply_chain_impacts(
-        country_list,
-        hazard_list,
-        sector_list,
-        scenario,
-        ref_year,
-        n_sim_years,
-        io_approach,
-        save_by_country=False,
-        save_by_hazard=False,
-        save_by_sector=False,
-        seed=1312,
-        indirect_output_dir="results/indirect",
-        direct_output_dir="results/direct",
-        business_interruption=True,
-        calibrated=True
-):
+DO_PARALLEL = True
+
+DO_DIRECT = True       # Calculate any direct impacts that are missing based on the config
+DO_YEARSETS = True     # Calculate any direct impact yearsets that are missing based on the config
+DO_MULTIHAZARD = False  # Also combine hazards in each calculation year to shock the supply chain
+DO_INDIRECT = True      # Calculate any indirect impacts that are missing based on the config
```

Comment:
Maybe it makes sense to expose these variables as arguments? Such that we don't have to change the code everytime we want to run a certain configuration. Alternatively, I think it would also be OK to provide them in the config, I do understand your reasoning though for separating it

Reply:
Raised in an issue here: https://github.com/ChrisFairless/nccs-supply-chain/issues/40